### ISSUE TITLE
IDE0052 Reported for Ref Field Assignment

### ISSUE THREAD

**Issue by ThadHouse** [dotnet/roslyn#71847]
C# 12 / .NET 8

Assigning to a ref field does not suppress the IDE0052 warning for the ref field being assigned to but never read from. Even though the whole point of the ref field is to assign to other memory.

**Steps to Reproduce**:

1. Compile the following ref struct

```
public ref struct TestStruct {
    private ref readonly byte m_toPin;
    private ref IntPtr m_toAssignPin;

    private IntPtr value;

    public void FromManaged(Span<byte> toPin) {
        m_toPin = ref toPin.GetPinnableReference();
        m_toAssignPin = ref Unsafe.AsRef(ref value);
    }

    public void ToUnmanaged() {
        m_toAssignPin = (IntPtr)Unsafe.AsPointer(ref Unsafe.AsRef(in m_toPin));
    }
}
```

2. See that m_toAssignPin reports an IDE0052 warning, even though the assignment in ToUnmanged() writes through the ref, and removing it wouldn't be possible (In my actual code the ref address assigned to m_toAssignPin is much more complex to find then this example).

**Diagnostic Id**:

IDE0052

**Expected Behavior**:

No IDE0052 Warning

**Actual Behavior**:

IDE0052 is reported.
